Documentation/General/productPlanning/requirements.tex

```diff
@@ -0,0 +1,146 @@
+	\begin{itemize}
+	\item The song received $\frac{\$number\_of\_users\_in\_room}{3}$ votes to skip.
+	\item The song received at least 10 votes, and the ratio upvote:downvote is 1:2 or less.
+	\item A moderator skips the song (to prevent escalations)
+	\end{itemize}
+	
+\item The system shall periodically suggest the user to hop to another room to encourage more interaction and we can direct the user to a more preferred mood.
+
+\item The multi-layer perceptron shall be trained with metadata provided by EEMCS
+
+\item The upvotes and downvotes for a song change the weights of the multi-layer perceptron.
+
+\item The user shall be rewarded for up- and downvoting with "treats". These treats are achievement points.
+
+\item The system shall retrieve more metadata about a song from AcousticBrainz\footnote{\url{http://acousticbrainz.org/}}.
+
+\item If metadata is not present in the AcousticBrainz database the system shall run an analysis itself using the AcousticBrainz software.
+
+\end{itemize}
+
+\subsubsection{Could haves}
+\begin{itemize}
+\item The system shall provide a search engine that aids the user in finding a song to add.
+
+\item The web interface shall notify the user if they made an illogical combination of moods, warning them results may vary.
+
+\item If SoundCloud provides MoodCat with a purchase\_url \footnote{\url{https://developers.soundcloud.com/docs/api/reference\#tracks}}, This will be made available to the user.
+
+\item With enough achievement points the user can receive predefined perks, such as the ability to add a number to the room.
+
+
+\end{itemize}
+
+\subsubsection{Would/Won't haves}
+\begin{itemize}
+\item The service won't have any other front end than the web interface.
+
+\item The web interface won't have music visualization.
+
+\item The service won't stream the music files itself, instead the SoundCloud API is used.
+
+\item The service won't have social network integration.
+
+\item The chat won't have a chatbot.
+
+\item The service won't have a list of listeners.
+
+\item The service won't have a Google scraper to display trivia.
+
+\item The users won't have the ability to comment on songs.
+
+\item The service won't have trending topics or moods.
+
+\end{itemize}
+
+\subsection{Non-functional requirements}
+\begin{itemize}
+
+\item For the project, the Scrum methodology shall be applied.
+
+\item Code shall be versioned using Git
+
+\item Git shall be used with the PR methodology: Every new feature will be developed in a branch, once completed it will be turned into a pull request. This will be reviewed and improved, and if it's accepted the branch gets merged into master.
+
+\item Frontend and Backend shall communicate in a RESTful way, using an API.
+
+\item The product shall consist of a seperate frontend and backend repo, ensuring proper RESTful implementation is used.
+
+\item The backend will be implemented in Java. The frontend web app will be implemented in AngularJS. 
```

Comment:
AngularJS is not a language but a framework. The frontend will be developed using web techniques as HTML, CSS and Javascript. We will use the AngularJS framework. But it's in no way a requirement.